### PR TITLE
Enable Datadog Real User Monitoring for Publish frontend

### PR DESCRIPTION
**Pull request by tiggreen**
Enable [Real User Monitoring](https://docs.datadoghq.com/real_user_monitoring/) to Publish frontend. We want to give this product a try starting from Publish. We have 14 days of free trial to evaluate and estimate the cost of this product.

## Description
Quote:

**What is Real User Monitoring?**
> Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real-time activities and experiences of individual users to prioritize engineering work on the features with the highest business impact. The datadog-rum library supports all modern desktop and mobile browsers. Resources collection is limited on IE10 and IE11.

## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/server/index.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { datadogRum } from '@datadog/browser-rum';
```
Comment (tiggreen) [inline packages/server/index.js:1]: Is this okay?

Comment (tiggreen) [reply packages/server/index.js:1]: Oh actually it's failing. I should push a fix.

File: packages/server/index.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+const datadogRum = require('@datadog/browser-rum');
```
Comment (tiggreen) [inline packages/server/index.js:1]: Is this okay?

**Comment by hamstu** [comment]
Hey @tiggreen! Thanks for taking a spike on this.

I think instead of putting this in the `server/index.js` file, this would actually need to be bundled into our frontend (as RUM appears to be frontend based tracking.). 

So you'd go back to using the `ìmport` syntax, and put it in our web entrypoint file: https://github.com/bufferapp/buffer-publish/blob/ccf7af670e644f92dc5915b00d0618e9129b23d3/packages/web/index.jsx#L1

I'm actually happy to make this change next week if you like, just let me know! This sounds like a very cool feature try out 😄

File: packages/server/index.js (hunk 2 above)
Comment (hamstu) [reply packages/server/index.js]: See my comment re: putting this in the web entrypoint instead of the server :)

**Comment by tiggreen** [comment]
Hey @hamstu that would be amazing if you could help me set this up for this repo. We have only 14 days left for the trial so if you have time today or tomorrow that would be so great <3 

It looks like Federico successfully enabled it to Analyze frontend in [this PR](https://github.com/bufferapp/buffer-analyze/pull/726).

Thank you for being so on top of things, friend! 🤗 🌮 🌮

**Comment by tiggreen** [comment]
In the meantime, @hamstu I will add the keys to the production secret! :)

**Comment by tiggreen** [comment]
Thanks again @hamstu! Is it safe to deploy this even though some checks are red?

**Comment by hamstu** [comment]
Just re-running the e2e tests @tiggreen - we've had some flakiness their since they are new, so it's not related. Will test it once those pass and then I can merge this.

**Comment by hamstu** [comment]
Ok, looks like it's working @tiggreen! ✅  Gonna merge it in.

**Pull request merged**